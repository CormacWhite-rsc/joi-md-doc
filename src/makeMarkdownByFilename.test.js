const makeMarkdownByFilename = require('./makeMarkdownByFilename');
const rootSchema = require('./../rootSchema');

describe(`makeMarkdownByFilename`, () => {
  describe(`unit`, () => {
    it(`should support primitive type`, () => {
      const joiSchema = {
        describe: () => ({
          metas: [{ name: 'Test', filename: 'test' }],
          keys: {
            foo: { type: 'number' },
            bar: { type: 'string' },
            baz: { type: 'boolean' },
          },
        }),
      };

      expect(makeMarkdownByFilename(joiSchema)).toMatchSnapshot();
    });
  });

  describe(`integration`, () => {
    it(`should match snapshot`, () => {
      expect(makeMarkdownByFilename(rootSchema)).toMatchSnapshot();
    });
  });
});
