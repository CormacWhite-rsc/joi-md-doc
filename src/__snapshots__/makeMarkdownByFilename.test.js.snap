// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeMarkdownByFilename integration should match snapshot 1`] = `
Object {
  "level1Schema": "[rootSchema](./rootSchema.md)
# Level 1 Schema

## foo

> \`string\` | Optional




## bar

> \`string\` | Optional




## [level2Schema](./level2Schema.md)

> \`object\` | Optional



",
  "level2Schema": "[rootSchema](./rootSchema.md) / [level1Schema](./level1Schema.md)
# Level 2 Schema

## baz

> \`string\` | Optional



",
  "rootSchema": "# Root Schema

## version

> \`number\` | Optional | Default: \`1.0.0\`


Follows semantic versioning.

## group

> \`number\` | Required




## type

> \`string\` | Required




## [level1Schema](./level1Schema.md)

> \`object\` | Optional



",
}
`;

exports[`makeMarkdownByFilename unit should mark required fields 1`] = `
Object {
  "test": "# Test

## foo

> \`number\` | Required



",
}
`;

exports[`makeMarkdownByFilename unit should support default meta tags 1`] = `
Object {
  "test": "# Test

## foo

> \`number\` | Optional | Default: \`1\`



",
}
`;

exports[`makeMarkdownByFilename unit should support descriptions 1`] = `
Object {
  "test": "# Test

## foo

> \`number\` | Optional


This is a description.
",
}
`;

exports[`makeMarkdownByFilename unit should support nested schemas/objects 1`] = `
Object {
  "child": "[parent](./parent.md)
# Child

## foo

> \`number\` | Optional



",
  "parent": "# Parent

## [foo](./foo.md)

> \`object\` | Optional



",
}
`;

exports[`makeMarkdownByFilename unit should support primitive type 1`] = `
Object {
  "test": "# Test

## foo

> \`number\` | Optional




## bar

> \`string\` | Optional




## baz

> \`boolean\` | Optional



",
}
`;
